library/ssl_srv.c

```diff
@@ -1700,8 +1700,8 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
              | ( buf[ciph_offset + 1]      );
 
     if( ciph_len < 2 ||
-        ciph_len + 2 + ciph_offset + 1 > msg_len || /* 1 for comp. alg. len */
-        ( ciph_len % 2 ) != 0 )
+        (size_t) ciph_len + 2 + ciph_offset + 1 > msg_len || /* 1 for comp. alg. len */
```

Comment:
Won't the cast to a larger type here undo the saving by declaring a smaller type?